### ISSUE TITLE
Timeout Issues - 5 Minute Downtime only on "selected" connections

### ISSUE THREAD

**Issue by michacassola** [WordOps/WordOps#310]
Since a few month I have been trying to figure out some network problems that are related to timeouts (522 on Cloudflare). The connection goes down for 5 minutes but only for my connection or for one uptime monitoring service (not all of them that I use!).

Can we increase the `keepalive_timeout 8;` to `keepalive_timeout 16`; please here please?
https://github.com/WordOps/WordOps/blob/master/wo/cli/templates/nginx-core.mustache
Here a source that also specifies some other tweaks:
https://www.digitalocean.com/community/tutorials/how-to-optimize-nginx-configuration
```
client_body_timeout 12;
client_header_timeout 12;
keepalive_timeout 15;
send_timeout 10;
```

Or am I totally off with these suggestions?

Also I am checking out GridPanes NGINX configs and they have some other tweaks in place:
https://gitlab.gridpane.net/gp-public/nginx-configs/-/blob/master/nginx.conf
`worker_rlimit_nofile 789022;`
Maybe we should increase from 100000 to 300000?

Also I got a Bad Gateway Error once which I since suspect has also maybe something to do with my problems.
Also another subject goes this way: https://community.wordops.net/d/615-here-is-how-i-fix-the-error-502-bad-gateway/14
Could we take GridPane as an example again and increase to
```
fastcgi_buffers 8 64k;
fastcgi_buffer_size 128k;
fastcgi_busy_buffers_size 256k;
```
?
https://gitlab.gridpane.net/gp-public/nginx-configs/-/blob/master/conf.d/fastcgi.conf
Is fastcgi even used in connection with redis page caching?

PS: How and where do I commit these kind of changes if I wanted to do that?

**Comment by Gfxfree** [comment]
You can try disable option: Always Online™ in Cloudflare.

**Comment by emanweb** [comment]
@michacassola Have you figured this out? I am having same problem too. It only happens to a very few cases. Please let me know.

**Comment by michacassola** [comment]
I suspect it was the rate limiting in nginx after all. I was using Oxygen Builder which is known to hammer the ajax admin endpoint and therefore trip up the rate limiting. Others like SmartSlider also do it I guess.
I have moved on from Oxygen and SmartSlider to Livecanvas.
Also I have made my own hosting script, which is still in the works but very close to 1.0: https://github.com/HOSTYON/qycli 
I did that because none of the options I used so far were opinionated enough and I use Cloudflare heavily for everything, so it integrates with Cloudflare a lot. There are some other reasons I started it but that is the main one.
